geonode_mapstore_client/client/js/components/home/FiltersMenu.jsx

```diff
@@ -12,20 +12,54 @@ import ReactResizeDetector from 'react-resize-detector';
 import Message from '@mapstore/framework/components/I18N/Message';
 import FaIcon from '@js/components/home/FaIcon';
 import useLocalStorage from '@js/hooks/useLocalStorage';
+import { filterMenuItems } from '@js/utils/MenuUtils';
 
 const FiltersMenu = forwardRef(({
     formatHref,
     orderOptions,
     order,
+    actions,
     filters,
     style,
     onClick,
     layoutSwitcher,
-    defaultLabelId
+    defaultLabelId,
+    user
 }, ref) => {
 
     const selectedSort = orderOptions.find(({ value }) => order === value);
     const [cardLayoutStyle] = useLocalStorage('layoutCardsStyle');
+    const state = {
+        user
+    };
+
+    const createNewElement = (actions?.authenticated && state.user) && (actions?.options.length > 0 && <Dropdown alignRight>
```

Comment:
Instead use a Dropdown component for a single action we could use an approach similar to MenuIndex, BrandNavbar or Footer